lighthouse-core/audits/audit.js

```diff
@@ -125,6 +125,62 @@ class Audit {
     };
   }
 
+  /** @typedef {{
+   * content: string;
+   * title: string;
+   * highlights: LH.Audit.DetailsRendererSnippetHighlight[];
+   * node?: LH.Audit.DetailsRendererNodeDetailsJSON;
+   * maxLineLength?: number;
+   * maxLinesAroundHighlight?: number;
+   * }} SnippetInfo */
+  /**
+   * @param {SnippetInfo[]} snippetInfos
+   * @return {LH.Audit.DetailsRendererDetailsJSON}
+   */
+  static makeSnippetListDetails(snippetInfos) {
+    return {
+      type: 'snippet-list',
+      items: snippetInfos.map(({
+        content,
+        title,
+        highlights,
+        node,
+        maxLineLength = 200,
+        maxLinesAroundHighlight = 20,
+      }) => {
+        const allLines = Audit._makeSnippetLinesArray(content, maxLineLength);
+        const lines = Util.filterRelevantLines(allLines, highlights, maxLinesAroundHighlight);
+
+        return {
+          lines,
+          title,
+          highlights,
+          lineCount: allLines.length,
+          node,
+        };
+      })};
+  }
+
+  /**
+   * @param {string} content
+   * @param {number} maxLineLength
+   * @returns {LH.Audit.DetailsRendererSnippetLine[]}
+   */
+  static _makeSnippetLinesArray(content, maxLineLength) {
+    return content.split('\n').map((line, lineIndex) => {
+      const lineNumber = lineIndex + 1;
+      /** @type LH.Audit.DetailsRendererSnippetLine */
```

Comment:
seems like this isn't needed?

Reply:
Without this TS complains about adding the `truncated` property.